packages/main/src/plugin/kubernetes-exec-transmitter.ts

```diff
@@ -0,0 +1,119 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { WritableOptions } from 'stream';
+import { Writable, Readable } from 'stream';
+
+export const DEFAULT_COLUMNS: number = 80;
+export const DEFAULT_ROWS: number = 60;
+
+export interface TerminalSize {
+  height: number;
+  width: number;
+}
+
+export class ExecStreamWriter extends Writable {
+  protected transmitter: Writable;
+
+  constructor(transmitter: Writable, options?: WritableOptions) {
+    super(options);
+    this.transmitter = transmitter;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  _write(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null) => void) {
+    this.transmitter._write(chunk, encoding, callback);
+  }
+}
+
+export class ResizeableTerminalWriter extends ExecStreamWriter {
+  protected columns: number;
+  protected rows: number;
+
+  constructor(
+    transmitter: Writable,
+    terminalSize: TerminalSize = { width: DEFAULT_COLUMNS, height: DEFAULT_ROWS },
+    options?: WritableOptions,
+  ) {
+    super(transmitter, options);
+    this.columns = terminalSize.width;
+    this.rows = terminalSize.height;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  _write(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null) => void): void {
```

Comment:
i believe we should be using `unknown` instead of `any`  in new code, right @benoitf ?

Reply:
yes avoid any

Reply:
fixed